server/plugins/api/Twitter.ts

```diff
@@ -1,22 +1,79 @@
 import { APIReturn } from "./APIReturn";
 import { GenericNotifier } from "./GenericNotifier";
 
+import * as TwitterAPI from "twit";
+
 export default class Twitter implements GenericNotifier<{}> {
 
+	private consumerKey: string;
+	private consumerSecret: string;
+	private accessToken: string;
+	private accessTokenSecret: string;
+
+	constructor() {
+		const consumerKey = process.env.TWITTER_CONSUMER_KEY;
+		const consumerSecret = process.env.TWITTER_CONSUMER_SECRET;
+		const accessToken = process.env.TWITTER_HACKGT_ACCESS_TOKEN;
+		const accessTokenSecret = process.env.TWITTER_HACKGT_ACCESS_TOKEN_SECRET;
+
+		if (!consumerKey) {
+			console.error("Missing Twitter consumer_key env var");
+		}
+		if (!consumerSecret) {
+			console.error("Missing Twitter consumer_secret env var");
+		}
+		if (!accessToken) {
+			console.error("Missing Twitter access_token env var");
+		}
+		if (!accessTokenSecret) {
+			console.error("Missing Twitter access_token_secret env var");
+		}
+
+		if (!consumerKey || !consumerSecret || !accessToken || !accessTokenSecret) {
+			throw new Error("Twitter env vars missing, aborting...");
+		}
+
+		this.consumerKey = consumerKey;
+		this.consumerSecret = consumerSecret;
+		this.accessToken = accessToken;
+		this.accessTokenSecret = accessTokenSecret;
+	}
+
 	public async setup(): Promise<void> {
-		return;
+		return; // No setup currently
 	}
 
 	public async check(config: any): Promise<{}> {
-		return {};
+		return {}; // No config currently
 	}
 
 	public async sendMessage(message: string, config: {}): Promise<[APIReturn]> {
-		return [{
-			error: false,
-			key: "Twitter",
-			message: "Twitter success"
-		}];
+		const client = new TwitterAPI({
+			consumer_key: this.consumerKey,
+			consumer_secret: this.consumerSecret,
+			access_token: this.accessToken,
+			access_token_secret: this.accessTokenSecret
+		});
+		const params = { status: message };
+		const res: Promise<APIReturn> = new Promise((resolve, reject) => {
+			client.post('statuses/update', params, (error, data: any, response) => {
```

Comment:
We still can't have an `any` type since we lose all type checking here.
What if `data.user.screen_name` doesn't exist and the plugin crashes _after_ its already successfully sent a tweet?
That would be a bad situtation, best thing is either keep it as `object` and do manual checks or remove this message.

Reply:
Can I leave it as any and check for existence of every property I use or should I assert any for all the times I try to reference a variable (including in the checks)